### ISSUE TITLE
Handling of POSIXct classes in get_bars 

### ISSUE THREAD

**Issue by yogat3ch** [yogat3ch/AlpacaforR#3]
Hey again! 
In an attempt to get recent prices I passed the following to get_bars:
```
price <- AlpacaforR::get_bars(s, from = {lubridate::now() - lubridate::minutes(5)}, to = lubridate::now(), timeframe = "minute")
```
and received the following error:
```
Error: parse error: trailing garbage
                                   400 Bad Request
                     (right here) ------^
```
I think this was due to the format of the from & to arguments, I suspected this because using strftime I was able to return a response that was more than I was looking for. It worked returning the previous 1000 minutes rather than 5, but I imagine that must be an API default?
Since we're now using lubridate in the package, it's possible to account for these exceptions that occur when POSIXct classes are provided for the from & to args with the following addition to the if statement regarding timeframes other than "day":
```
if(!(timeframe == "1D" | timeframe == "day") & (lubridate::is.POSIXct(from) | lubridate::is.POSIXct(to))){
  from <- strftime(from, "%Y-%m-%dT%H:%M:%S%z", tz = Sys.timezone())
  to <- strftime(to, "%Y-%m-%dT%H:%M:%S%z", tz = Sys.timezone())
} else if(!(timeframe == "1D" | timeframe == "day")){
  from = paste0(from,stringr::str_extract(format(Sys.time(), "%Y-%m-%dT%H:%M:%OS%z"), "T.*"))
  to = paste0(to,stringr::str_extract(format(Sys.time(), "%Y-%m-%dT%H:%M:%OS%z"), "T.*"))
} else {
  from = paste0(from,"T20:30:00-04:00")
  to = paste0(to,"T20:30:00-04:00")
}
```

I think this will account for that exception when using POSIXct that aren't ISO formatted as the API specifies.

In service,
Stephen

**Comment by jagg19** [comment]
Hi @yogat3ch,

I played around with different formats, and it does seem like the format was the issue. I never considered if the user was already supplying a POSIXct format for both from and to arguments, thanks!! This addition works great.

As far as the result returning 1000 minutes, if you want the last 5 minutes then you can set limit = 5 to get the last 5 minutes. Alpaca's [/bars endpoint](https://docs.alpaca.markets/api-documentation/web-api/market-data/bars/) allows a limit parameter, which sets the number of recent price bars to return. So something like this would return the last 5 minutes for a stock;

`AlpacaforR::get_bars("AMZN",timeframe="minute",limit = 5)`

If limit is NULL, I set the default to return 1000 for the minute timeframe, because the API will return an error if more than 1000 is requested (1000 is the max on Alpaca). It looks like #4 is related to get_bars so I'll check that out before pushing any updates. Thanks again!

**Comment by yogat3ch** [comment]
Hi @jagg19,
>As far as the result returning 1000 minutes, if you want the last 5 minutes then you can set limit = 5 to get the last 5 minutes. Alpaca's /bars endpoint allows a limit parameter, which sets the number of recent price bars to return. So something like this would return the last 5 minutes for a stock;

Good to know, thanks for the info! I'll update accordingly.

This is resolved by #4 so we're good to go here. 
Thanks!
